operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/EventSourceManager.java

```diff
@@ -25,38 +25,25 @@
 
   private final ReentrantLock lock = new ReentrantLock();
   private final Set<EventSource> eventSources = Collections.synchronizedSet(new HashSet<>());
-  private EventProcessor<R> eventProcessor;
-  private TimerEventSource<R> retryAndRescheduleTimerEventSource;
+  private final EventProcessor<R> eventProcessor;
   private ControllerResourceEventSource<R> controllerResourceEventSource;
+  private final Controller<R> controller;
 
-  EventSourceManager() {
-    init();
+  EventSourceManager(EventProcessor<R> eventProcessor) {
+    this.eventProcessor = eventProcessor;
+    controller = null;
   }
 
   public EventSourceManager(Controller<R> controller) {
-    init();
+    this.controller = controller;
     controllerResourceEventSource = new ControllerResourceEventSource<>(controller);
+    this.eventProcessor = new EventProcessor<>(this);
```

Comment:
The event source manager should not instantiate the processor. I that the controller manages the lifecycle is much cleaner. This is what was moved out as part of the design changes.

Reply:
This doesn't make things cleaner: look at how ugly the `setEventHandler` / `setEventManager` mess is.

Reply:
except for those :D